src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+            threshold = (int) (table.length * LOAD_FACTOR);
+        }
+        putVal(hash(key), key, value);
+    }
 
+    private void putVal(int hash, K key, V value) {
+        int currentPosition = hash % (table.length - 1);
+        Node<K, V> tempNode = new Node<>(hash, key, value, null);
+        if (table[currentPosition] == null) {
+            table[currentPosition] = tempNode;
+            size++;
+        } else {
+            Node<K, V> iterationNode = table[currentPosition];
+            while (iterationNode.next != null || Objects.equals(iterationNode.key, tempNode.key)) {
+                if (Objects.equals(iterationNode.key, tempNode.key)) {
+                    iterationNode.value = tempNode.value;
+                    return;
+                }
+                iterationNode = iterationNode.next;
+            }
+            iterationNode.next = tempNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[hash(key) % (table.length - 1)];
```

Comment:
let's create method getIndex that will calculate index for given key - this way it will be more obvious

Reply:
done